From 5c70644070803b665f64c2d3212b55a4a3270765 Mon Sep 17 00:00:00 2001
From: Daniel Stone <daniel@fooishbar.org>
Date: Sun, 10 Mar 2013 11:29:38 -0700
Subject: [PATCH 4/4] EGL: Specify vec2d precision qualifier in fragment shader

GLSL does not specify a default precision qualifier here, so we have to
explicitly define it.  Use highp, as it gets redefined to mediump by
QGLShaderProgram when not available.

Signed-off-by: Daniel Stone <daniel@fooishbar.org>
Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
---
 src/plugins/platforms/qwayland-egl/qwaylandglcontext.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/plugins/platforms/qwayland-egl/qwaylandglcontext.cpp b/src/plugins/platforms/qwayland-egl/qwaylandglcontext.cpp
index 4d8017d..a0ac6d9 100644
--- a/src/plugins/platforms/qwayland-egl/qwaylandglcontext.cpp
+++ b/src/plugins/platforms/qwayland-egl/qwaylandglcontext.cpp
@@ -124,7 +124,7 @@ void QWaylandGLContext::swapBuffers(QPlatformSurface *surface)
                                                                             gl_Position = position;\n\
                                                                             outTexCoords = texCoords.xy;\n\
                                                                         }");
-            m_blitProgram->addShaderFromSourceCode(QOpenGLShader::Fragment, "varying vec2 outTexCoords;\n\
+            m_blitProgram->addShaderFromSourceCode(QOpenGLShader::Fragment, "varying highp vec2 outTexCoords;\n\
                                                                             uniform sampler2D texture;\n\
                                                                             void main()\n\
                                                                             {\n\
-- 
1.8.2.1

